src/containers/SubscriberSignup/index.js

```diff
@@ -103,32 +152,48 @@ class SubscriberSignup extends React.Component {
                 {getFieldDecorator('districts', {
                     rules: [{ required: true, message: 'Must have districts' }],
                 })(
-            <Select
-                mode="tags"
-                style={{ width: '200px' }}
-                onChange={this.handleChange}
+                <Select
+                    mode="tags"
+                    style={{ width: '200px' }}
                 >
                     {options}
                 </Select>
             )}
             </Form.Item>
             <Form.Item>
-            <Button
-                type="primary"
-                htmlType="submit"
-                disabled={this.hasErrors(getFieldsError())}
-            >
-                Submit
-            </Button>
+                {getFieldDecorator('key')(
+                    <Input type="hidden" />
+                )}
+            </Form.Item>
+            <Form.Item>
+                <Button
+                    type="primary"
+                    htmlType="submit"
+                    disabled={this.hasErrors(getFieldsError())}
+                >
+                    {submitText}
+                </Button>
+                <Button
+                    type="default"
+                    htmlType="reset"
+                    onClick={this.clearForm}
+                >
+                    Clear
+                </Button>
             </Form.Item>
         </Form>
         );
     }
 }
 
 const mapDispatchToProps = dispatch => ({
-    submitSubscriber: (person) => dispatch(userStateBranch.actions.submitSubscriber(person)),
+    submitSubscriber: (person) => dispatch(subscriberStateBranch.actions.submitSubscriber(person)),
+    getAllSubscribers: () => dispatch(subscriberStateBranch.actions.getAllSubscribers()),
```

Comment:
to make this not sound like a selector, consider naming this something like, `requestAllSubscribers` or `startSubscriberRequest`

Reply:
Done